docs/smart-contracts/accounts-keys.md

```diff
@@ -0,0 +1,160 @@
+For example, you might want to create an account with a name that identifies it as a personal or primary account, for example, Lola-Pistola, so that it's easy to differentiate it from an account that you own jointly with another party or a group, for example, Las-Pistolas.
+
+However, using arbitrary account names like these examples can make your account vulnerable to certain kinds of attacks. 
+For example, an attacker might try to hijack a transaction that creates an account or transfers funds by changing the keyset associated with the account name. 
+One way to prevent an account from being hijacked by an attacker is to create **principal** account. 
+A principal is a way to enforce a one-to-one relationship between a **guard** and what the guard is there to protect, like the ownership of an account balance. 
+If an attacker tries to hijack the account by changing the keyset—the most common type of guard—the keyset won't match the one defined in the underlying ledger, so the transaction would fail.
+
+### Keysets and principals
+
+Keysets represent the most-commonly used type of guard and are the most similar to how most blockchains protect access to accounts using public and secret keys. 
+As you've already seen, a keyset holds a collection of one or more public keys and a predicate function that defines how many of those keys must sign to authorize an action. 
+In Pact, the guard is a Boolean value that must return true for an associated action to take place.
+
+By default, when you define a keyset with a **single** public key and the **keys-all** predicate, the result is an account name that starts with the `k:` prefix, followed by the public key for the account. This naming convention creates a principal account for an individual key.
+
+You can also create principal accounts for keysets that have multiple keys and that use either built-in or custom predicate rules. 
+For example, you can define a keyset with two public keys and the built-in predicate `keys-any`. 
+If you create a principal for this account, the keyset information is used to generate a unique hash and the account is created using the `w:` prefix, followed by the hash for the guard.
+
+### Other types of guards
+
+In addition to keysets, Pact supports several other types of guards that you can implement in smart contracts. 
+These guards give you flexibility for handling different types of authorization scenarios, and in general, you can use any type of guard to create a principal account by generating a unique hash.
+
+Each type of principal and guard uses a unique prefix, so they are easy to recognize. 
+If you create principal accounts for the guards, you'll see that the principal accounts use the following set of prefixes:
+
+- k: for single key keysets
+- w: for multiple keys keysets
+- u: for user guards
+- c: for capability guards
+
+To learn more about different types of guards and how to use them, see [Guards](/smart-contracts/guards). 
```

Comment:
So reaching near the end of the principals section, it's actually never discussed how they are created. Can we attach a simple example of `create-principal` our docs, to show what it looks like?